### ISSUE TITLE
NPE crash ticking player on world load

### ISSUE THREAD

**Issue by kitsu** [SilentChaos512/ExtendedDays#9]
I am having an intermittent crash issue with extended days.
Reading the exception I have the vague impression that you are trying to interact with the player before it is fully loaded?

I've attached the latest log:
[crash-2018-01-14_07.19.38-server.txt](https://github.com/SilentChaos512/ExtendedDays/files/1629465/crash-2018-01-14_07.19.38-server.txt)

I will attempt to recreate the crash without OptiFine.

EDIT: After restarting multiple times without OptiFine I can't recreate the issue, and after upgrading to Forge 2587 and the latest OptiFine I also can't recreate the issue.

That said I am not ready to blame OptiFine.
I've configured Extended Days to add 1 Minute every hour.
I can see using the Journey Map clock HUD and the Bibliocraft clock when time freezes.
Once I got logged in I can see that I must have last logged off at exactly 1:00 Minecraft time.
Is it possible that time-stretching during world load/client connect could cause the NPE?

EDIT2: And then it comes back, disable OptiFine and it loads fine, re-enable and it loads fine...
Reading the new crash log I see it is checking my "Vehicle" which is null?
Let me know if you further testing to narrow down the cause.

### Versions

Minecraft Version: 1.12.2
Forge: 14.23.1.2577
Extended Days Version: 0.1.1 AND 0.2.0
Silent Lib Version: 2.2.16

**Comment by SilentChaos512** [comment]
Not sure about the cause, but some simple null checks should fix the issue. That particular block of code can wait a little while if it needs to, it just replaces the sky renderer.

**Comment by ElvenChaos** [comment]
I'm not sure if my issue is the same as this one, but I am also getting a NPE ticking player on world load. In my example, I am trying to create a new world, and after creation (but before the player actually loads into the world) the NPE occurs. Notably my instance does not use Optifine, and I am on Forge 2587.

https://pastebin.com/3j79yGYN

**Comment by SilentChaos512** [comment]
Version 0.2.1 should be available to download sometime soon. Please reopen the issue or comment if it still crashes.

**Comment by ElvenChaos** [comment]
https://pastebin.com/KUJXeawZ

Still crashes.

**Comment by SilentChaos512** [comment]
Sorry, hard to fix issues I can't reproduce. Try this one: 
[ExtendedDays-1.12-0.2.1b-13.zip](https://github.com/SilentChaos512/ExtendedDays/files/1633291/ExtendedDays-1.12-0.2.1b-13.zip). **Extract the JAR from the ZIP file first.** I'll upload to Curse if it works.

**Comment by ElvenChaos** [comment]
Before you posted that, I turned off the sky rendering from this mod and that prevented crashes (i.e. going back to "pause and play"). After I used 0.2.1b-13 and re-enabled the special sky rendering, there were no more crashes.

Also, I think I might have figured out the offending mod-combination due to this: Dynamic Surroundings, with the auroras in particular being an issue. Since the crashing is fixed and I need to investigate more, I will open up a new issue for the mod compatibility afterwards.
